docs/debugging.md

```diff
@@ -1,72 +1,92 @@
-
-```rust
-    let mut lexer = Lexer::new(src);
-    lexer.lex().expect("lexing failed");
-    let tokens = lexer.tokens;
-    dbg!(&tokens);
-    ...
-```
-Seeing the order of tokens can be a big help to understanding what the parser is working with.
-
-**Note:** flags `--dump-tokens` and `--dump-ast` are mutually exclusive. When using the flag `--dump-tokens`, the code will not be executed.
+**Note:** flags `--dump-tokens` and `--dump-ast` are mutually exclusive. When
+using the flag `--dump-tokens`, the code will not be executed.
 
-## Expressions
+## AST nodes
 
-Assuming the tokens looks fine, the next step is to see the AST.
-You can use the `boa_cli` command-line flag `--dump-ast`, which can optionally take a format type. Supports these formats: `Debug`, `Json`, `JsonPretty`. By default it is the `Debug` format.
+Assuming the tokens looks fine, the next step is to see the AST. You can use
+the `boa_cli` command-line flag `--dump-ast`, which can optionally take a
+format type. Supports these formats: `Debug`, `Json`, `JsonPretty`. By default
+it is the `Debug` format.
 
 Dumping the AST of a file:
 ```bash
 cargo run -- test.js --dump-ast # AST dump format is Debug by default.
 ```
+
 or with interactive mode (REPL):
 ```bash
 cargo run -- --dump-ast # AST dump format is Debug by default.
 ```
-Or manually, you can output the expressions in [forward](../boa/src/lib.rs#L36), add `dbg!(&expr);`
 
 These methods will print out the entire parse tree.
 
-**Note:** flags `--dump-tokens` and `--dump-ast` are mutually exclusive. When using the flag `--dump-ast`, the code will not be executed.
+**Note:** flags `--dump-tokens` and `--dump-ast` are mutually exclusive. When
+using the flag `--dump-ast`, the code will not be executed.
+
+## Compiler panics
+
+In the case of a compiler panic, to get a full backtrace you will need to set
+the environment variable `RUST_BACKTRACE=1`.
 
 ## Execution
 
-Once the tree has been generated [exec](../boa/src/lib.rs#L67) will begin to run through each expression. If the tokens and tree looks fine, you can start looking here.
-I usually just add `dbg!()` in the relevent places to see what the output is at the time.
+Once the tree has been generated [exec](../boa/src/lib.rs#L92) will begin to
+run through each expression. If the tokens and tree looks fine, you can start
+looking here. We usually just add `dbg!()` in the relevent places to see what
+the output is at the time.
 
 ## Debugger
 
 ### VS Code Debugger
 
-The quickest way to get debugging is to re-open the workspace in the container (using the Dockerfile provided). This is using the [Remote Containers plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers). Once inside make sure you have the CodeLLDB extension installed and add breakpoints.
+The quickest way to get debugging is to re-open the workspace in the container
```

Comment:
This bit isn’t true anymore, the debugger works on MacOS, Linux and windows so you don’t need the container for that. 

You could just say the “quickest way to get debugging is to use the CodeLLDB plugin and add breakpoints” more info https://jason-williams.co.uk/debugging-rust-in-vscode